src/components/ImgUpload.jsx

```diff
@@ -13,8 +13,27 @@ function ImgUpload(props) {
   const [uploadingPost, setUploadingPost] = useState(false);
   const imgInput = useRef(null);
   const [imagePreviews, setImagePreviews] = useState([]);
+  const [isValidimage, setisValidimage] = useState(true);
 
   const handleChange = (e) => {
+    const image = e.target.files[0];
+    if (!image) {
+      enqueueSnackbar("Select min 1 image!", {
+        variant: "error",
+      })
+      setisValidimage(false);
+      return false;
+    }
+    if (!image.name.match(/\.(jpg|jpeg|png)$/)) {
```

Comment:
you can add validation for the whole array of files

Reply:
I thought that too but it works. even if one of the file is of other types it will not expect it.

Reply:
I checked and it fails, logically as well as practically
![image](https://user-images.githubusercontent.com/77617189/208460310-ec2bcf8c-205c-4b26-bdea-e11991d4cacc.png)
first ones a jpg, other one gif, then 2 txt files, that's not a behavior we may want

Reply:
At my end I am not able to reproduce the problem

Reply:
https://user-images.githubusercontent.com/91470808/208461976-39b86e57-5e80-4591-a520-c4dbba447194.mp4

Reply:
when selecting the files select files of multiple types and first one of image type, then it'll be accepted.
Anyways you should see the logical error, that is, you're checking just the first one in array for type matching. you can instead use the any function to check if there's any invalid format in the upload and raise error

Reply:
> DummyGram.-.Google.Chrome.2022-12-19.21-00-04.mp4

about this, this works because files are sorted automatically by some condition and invalid file type came in the 0 index i.e., before the jpg but that won't always be the case

Reply:
Okay will try something and update you!